2020/TestRig/ServoCalibrate.java

```diff
@@ -0,0 +1,117 @@
+package TestRig;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.Disabled;
+import com.qualcomm.robotcore.hardware.Servo;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import team25core.GamepadTask;
+import team25core.Robot;
+import team25core.RobotEvent;
+import team25core.ServoCalibrateTask;
+
+@Autonomous(name = "ServoCalibrate")
+// @Disabled
+public class ServoCalibrate extends Robot {
+
+    List<Servo> servoList;
+    Map<Servo, Telemetry.Item> servoItemMap;
+    Iterator<Map.Entry<Servo, Telemetry.Item>> iterator = null;
+
+    Servo activeServo;
+    int numServos;
+
+    @Override
+    public void handleEvent(RobotEvent e)
+    {
+    }
+
+    protected String getServoName(Servo servo)
+    {
+        Set<String> names = hardwareMap.getNamesOf(servo);
+
+        /**
+         * Yes, this looks strange, but if a device has more than one name, we'll
+         * just use the first one.  In all practical usage I don't think there's
+         * ever more than one name.  And since Set is unordered there's no first()
+         * or last() operations.  Hence the funky non-iteration.
+         */
+        for (String name : names) {
+            return name;
+        }
+
+        return "NoName";
+    }
+
+    protected Servo getNextInMap()
+    {
+        if ((iterator == null) || (iterator.hasNext() == false)) {
+            iterator = servoItemMap.entrySet().iterator();
+        }
+
+        return iterator.next().getKey();
+    }
+
+    protected void clearMark(Telemetry.Item item)
+    {
+        item.setValue("");
+    }
+
+    protected void setMark(Telemetry.Item item)
+    {
+        item.setValue(" *");
+    }
+
+    @Override
+    public void init()
+    {
+        servoList = hardwareMap.getAll(Servo.class);
+        numServos = servoList.size();
+        servoItemMap = new LinkedHashMap<>();
+
+        if (numServos == 0) {
+            Telemetry.Item none = telemetry.addData("No servos", "");
+            return;
+        }
+
+        for (Servo servo : servoList) {
+            String name = getServoName(servo);
+            Telemetry.Item servoTelemetry = telemetry.addData(name, "");
+            servoItemMap.put(servo, servoTelemetry);
```

Comment:
for this for loop it searches through all the servos inside servoList and the String name gets the name of the servo. the name is the string caption and second part is object value